libraries/AP_InertialSensor/AP_InertialSensor_tempcal.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  IMU temperature calibration handling
+ */
+
+#include "AP_InertialSensor.h"
+
+#if HAL_INS_TEMPERATURE_CAL_ENABLE
+
+// temperature calibration parameters, per IMU
+const AP_Param::GroupInfo AP_InertialSensor::TCal::var_info[] = {
+
+    // @Param: ENABLE
+    // @DisplayName: Enable temperature calibration
+    // @Description: Enable the use of temperature calibration parameters for this IMU
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE", 1, AP_InertialSensor::TCal, enable,  0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: TMIN
+    // @DisplayName: Temperature calibration min
+    // @Description: The minimum temperature that the calibration is valid for
+    // @Range: -70 80
+    // @Units: degC
+    // @User: Advanced
+    // @Calibration: 1
+    AP_GROUPINFO("TMIN", 2, AP_InertialSensor::TCal, temp_min,  0),
+
+    // @Param: TMAX
+    // @DisplayName: Temperature calibration max
+    // @Description: The maximum temperature that the calibration is valid for
+    // @Range: -70 80
+    // @Units: degC
+    // @User: Advanced
+    // @Calibration: 1
+    AP_GROUPINFO("TMAX", 3, AP_InertialSensor::TCal, temp_max,  0),
+
+    // @Param: ACC1_X
+    // @DisplayName: Accelerometer 1st order temperature coefficient X axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: ACC1_Y
+    // @DisplayName: Accelerometer 1st order temperature coefficient Y axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: ACC1_Z
+    // @DisplayName: Accelerometer 1st order temperature coefficient Z axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+    
```

Comment:
Need to trigger on INS_MAX_INSTANCES here

Reply:
And all of the others below

Reply:
nope, those indexes are polynomial powers, not IMU indexes